### PR TITLE
Move Property Assignment Function into the functions package

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

Moves `PropertyAssignmentFunction` into the `functions` package so that it's more discoverable and reusable. Required removing some functionality from the function so that it could stay in the `conversions` package to avoid circular references.

Reuse of this function is required by WIP to create a `ConvertResourceFunction`, necessary for implementing the `Convertible` interface.

Harvested from PR #1533

**Prerequisites**
- [x] Merge PR #1606
- [x] Merge PR #1618 
- [ ] Rebase on `master`

**How does this PR make you feel**:
gif

**If applicable**:
- [x] this PR contains tests

### CODE REVIEW

File: hack/generator/pkg/conversions/property_conversions.go [hunk 1]
```diff
@@ -145,6 +145,15 @@ func CreateTypeConversion(
 	return nil, err
 }
 
+
+func NameOfPropertyAssignmentFunction(name astmodel.TypeName, direction Direction, idFactory astmodel.IdentifierFactory) string {
+	nameOfOtherType := idFactory.CreateIdentifier(name.Name(), astmodel.Exported)
+	return direction.SelectString(
+		"AssignPropertiesFrom"+nameOfOtherType,
+		"AssignPropertiesTo"+nameOfOtherType)
```
Comment (babbageclunk) [inline hack/generator/pkg/conversions/property_conversions.go:153]: Maybe clearer as:
```suggestion
	return "AssignProperties" + direction.SelectString("From", "To") + nameOfOtherType
```

Comment (babbageclunk) [reply hack/generator/pkg/conversions/property_conversions.go:153]: Actually maybe `.SelectString("From", "To")` could be built-in? Although that's not too bad as-is.

File: hack/generator/pkg/conversions/readable_conversion_endpoint.go [hunk 2]
```diff
@@ -60,3 +60,41 @@ func MakeReadableConversionEndpointForValueFunction(
 func (r ReadableConversionEndpoint) String() string {
 	return r.description
 }
+
+// Reader generates an expression to read our endpoint
```
Comment (babbageclunk) [inline hack/generator/pkg/conversions/readable_conversion_endpoint.go:64]:
```suggestion
// Read generates an expression to read our endpoint
```

**Review by babbageclunk — APPROVED** [review]

File: hack/generator/pkg/conversions/property_conversions.go (hunk 1 above)
Comment (theunrepentantgeek) [reply hack/generator/pkg/conversions/property_conversions.go:153]: I'm doing a Code Gardening PR that cleans up a few things like this, will include there.

File: hack/generator/pkg/conversions/readable_conversion_endpoint.go (hunk 2 above)
Comment (theunrepentantgeek) [reply hack/generator/pkg/conversions/readable_conversion_endpoint.go]: Fixed.

**Pull request merged**